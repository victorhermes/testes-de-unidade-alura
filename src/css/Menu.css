.menu {
    background-color: #2a5eec;
    height: 70px;
    top: 0;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
}

.menu-logo {
    width: 75px;
}
