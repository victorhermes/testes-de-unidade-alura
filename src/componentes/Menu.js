import React, { Component } from "react";
import { Link } from "react-router-dom";
import "../css/Menu.css";
import Logo from "../imgs/alura.svg";

export default class Menu extends Component {
    render() {
        return (
            <header className="menu">
                <Link to="/">
                    <img src={Logo} alt="Logo" className="menu-logo" />
                </Link>
            </header>
        );
    }
}
