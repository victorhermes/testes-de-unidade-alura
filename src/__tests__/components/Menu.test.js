import { shallow } from "enzyme";
import React from "react";

import Menu from "../../componentes/Menu";

describe("Testes do menu", () => {
    it("Deve haver o componente menu", () => {
        const componente = shallow(<Menu />);

        expect(componente.exists()).toBe(true);
    });

    it("Deve renderizar o menu", () => {
        const componente = shallow(<Menu />);

        expect(componente.find("Link")).toHaveLength(2);
        expect(componente.find(".link-produto").text()).toEqual(
            "Adicionar Produto"
        );
    });
});
