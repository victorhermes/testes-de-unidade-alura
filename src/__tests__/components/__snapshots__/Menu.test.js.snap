// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes do menu Deve renderizar o menu 1`] = `ShallowWrapper {}`;
